lib/mongoid/association/embedded/embeds_many/proxy.rb

```diff
@@ -235,8 +235,8 @@ def exists?
           # @example Finds the first matching document using a block.
           #   person.addresses.find { |addr| addr.state == 'CA' }
           #
-          # @param [ Array<Object> ] args Various arguments.
-          # @param [ Proc ] block Optional block to pass.
+          # @param [ Object... ] *args Various arguments.
+          # @yield &block Optional block to pass.
```

Comment:
Why is this a `@yield`?

Reply:
In my view the current way `@param [ Proc ] block` is not correct, because (a) it's not a standard arg, and (b) its not a Proc object.

Here's what [Yard officially says](https://www.rubydoc.info/gems/yard/file/docs/Tags.md#yield):

```
# For a block {|a,b,c| ... }
# @yield [a, b, c] Gives 3 random numbers to the block
def provide3values(&block)
  yield(42, 42, 42)
end
```

It seems they don't add document the `&block` parameter name, however, I think there is no harm in adding it to `@yield`. I would propose the following syntax:

```
# @yield [a, b, c] &block Gives 3 random numbers to the block
def provide3values(&block)
  yield(42, 42, 42)
end

# @yield [a, b, c] Gives 3 random numbers to the block
def provide3values
  yield(42, 42, 42)
end

# @yield [a, b, c] &block Gives 3 random numbers to the block
def provide3values(&block)
  block.call(42, 42, 42)
end
```